Polyhedron/demo/Polyhedron/Plugins/Point_set/Surface_reconstruction_plugin_impl.cpp

```diff
@@ -71,7 +71,10 @@ Polyhedron* poisson_reconstruct(const Point_set& points,
       return NULL;
     }
 
-    bool points_have_normals = (points.begin()->normal() != CGAL::NULL_VECTOR);
+    Point_set::const_iterator points_begin = (points.nb_selected_points() == 0
+                                              ? points.begin() : points.first_selected());
+
+    bool points_have_normals = (points_begin->normal() != CGAL::NULL_VECTOR);
```

Comment:
I just kept the same code without really thinking about it, but yes, I'll replace that.